model/lifecycle.go

```diff
@@ -409,6 +409,19 @@ func RestartVersion(versionId string, taskIds []string, abortInProgress bool, ca
 			event.LogTaskRestarted(t.Id, t.Execution, caller)
 		}
 	}
+
+	buildIdsMap := map[string]bool{}
+	var buildIds []string
+	for _, t := range tasksToRestart {
+		buildIdsMap[t.BuildId] = true
+	}
+	for buildId := range buildIdsMap {
+		buildIds = append(buildIds, buildId)
+	}
+	if err = UpdateVersionAndPatchStatusForBuilds(buildIds); err != nil {
+		return errors.Wrapf(err, "updating build and version status for version '%s'", versionId)
+	}
+
 	if err = build.SetBuildStartedForTasks(tasksToRestart, caller); err != nil {
```

Comment:
Would we need to call this before UpdateVersionAndPatchStatusForBuilds so that the builds are in the right state or?

Reply:
It doesn't matter because `UpdateVersionAndPatchStatusForBuilds` will only update the status of builds that have had their [statuses changed](https://gist.github.com/hadjri/f1287d20a311464513c465bf7767336f#file-task_lifecycle-go-L1353) by checking the state of the tasks (which have just been reset above).

Reply:
Ok, looks like they can be safely removed.

Reply:
Actually, only `setVersionStatus` can be safely removed because  `SetBuildStartedForTasks` performs an activation update that is necessary, which is something that `updateBuildStatus` does not do.